pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -1029,6 +1029,11 @@ type ObjectStoreSpec struct {
 	// +optional
 	// +nullable
 	HealthCheck BucketHealthCheckSpec `json:"healthCheck,omitempty"`
+
+	// Security represents security settings
+	// +optional
+	// +nullable
+	Security SecuritySpec `json:"security,omitempty"`
```

Comment:
Make it a pointer if optional.

Reply:
@leseb : I have added it as separate setting in `ObjectStoreSpec` not a reference from `ClusterSpec`. If it is a pointer to the `clusterSpec.Security` then changes made for this setting will affect `ClusterSpec.Security`? The new osd may end up using new settings than the old spec.

Reply:
I meant just do:

```
Security *SecuritySpec `json:"security,omitempty"`
```

Reply:
@leseb : Done